app/src/cards.rs

```diff
@@ -31,4 +32,21 @@ impl Cards for App {
             None => Err(CardCreateError::Unauthorized),
         }
     }
+
+    async fn card_search(&self, search: &str) -> Result<Vec<Card>, CardSearchError> {
+        let db = self.get::<Service<dyn Repository>>()?;
+
+        let (mut title_res, content_res, tags_res) = tokio::try_join!(
+            db.cards_find_by_title(search),
+            db.cards_find_by_content(search),
+            db.cards_find_by_tag(search)
+        )?;
+
+        title_res.extend(content_res);
+        title_res.extend(tags_res);
+
+        let search_results = title_res;
+
+        Ok(search_results.into_iter().unique_by(|c| c.id).collect())
```

Comment:
Might actually need to use some kind of search type indication or priority.

But for now let's just search everywhere